From 6893b274b640c4f691d6ed808b303543ba0bc6c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tadej=20Jane=C5=BE?= <tadej.j@nez.si>
Date: Mon, 13 Sep 2021 14:52:23 +0200
Subject: [PATCH 10/15] Explicilty import distutils.sysconfig subpackage in
 sitepackages_dir()

In general, submodules and subpackages are not imported when one imports
a package, unless they are explicitly included in the package's
__init__.py.

To avoid a potential issue, explicitly import get_python_lib() from
distutils.sysconfig subpackage.
---
 pew/pew.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pew/pew.py b/pew/pew.py
index 42e7a79..693e01a 100644
--- a/pew/pew.py
+++ b/pew/pew.py
@@ -371,8 +371,8 @@ def sitepackages_dir(env=os.environ.get('VIRTUAL_ENV')):
         sys.exit('ERROR: no virtualenv active')
     else:
         env_python = workon_home / env / env_bin_dir / 'python'
-        return Path(invoke(str(env_python), '-c', 'import distutils; \
-print(distutils.sysconfig.get_python_lib())').out)
+        return Path(invoke(str(env_python), '-c',
+            'from distutils.sysconfig import get_python_lib; print(get_python_lib())').out)
 
 
 def add_cmd(argv):
-- 
2.31.1

